src/Microsoft.AspNet.Mvc.Extensions/JsonResult.cs

```diff
@@ -81,80 +59,44 @@ public JsonResult(object value, IOutputFormatter formatter)
         public object Value { get; set; }
 
         /// <inheritdoc />
-        public override async Task ExecuteResultAsync([NotNull] ActionContext context)
+        public override Task ExecuteResultAsync([NotNull] ActionContext context)
         {
-            var objectResult = new ObjectResult(Value);
+            var response = context.HttpContext.Response;
 
-            // Set the content type explicitly to application/json and text/json.
-            // if the user has not already set it.
-            if (ContentTypes == null || ContentTypes.Count == 0)
+            var contentTypeHeader = ContentType;
+            if (contentTypeHeader == null)
             {
-                foreach (var contentType in DefaultContentTypes)
-                {
-                    objectResult.ContentTypes.Add(contentType);
-                }
+                contentTypeHeader = DefaultContentType;
             }
             else
             {
-                objectResult.ContentTypes = ContentTypes;
+                if (contentTypeHeader.Encoding == null)
+                {
+                    // 1. Do not modify the user supplied content type
+                    // 2. Parse here to handle parameters apart from charset
+                    contentTypeHeader = MediaTypeHeaderValue.Parse(contentTypeHeader.ToString());
+                    contentTypeHeader.Encoding = Encoding.UTF8;
```

Comment:
Technically it shouldn't.  Despite its widespread usage, the charset parameter is not defined for application/json.  https://tools.ietf.org/html/rfc7159#section-11